account_invoice_production_lot/models/account_move_line.py

```diff
@@ -19,11 +19,21 @@ class AccountMoveLine(models.Model):
     @api.depends("move_line_ids")
     def _compute_prod_lots(self):
         for line in self:
-            line.prod_lot_ids = line.mapped("move_line_ids.move_line_ids.lot_id")
+            lots = set()
+            stock_moves = line.sale_line_ids.mapped("move_ids")
+            for move in stock_moves:
+                for sml in move.move_line_ids:
+                    if sml.lot_id:
+                        lots.add(sml.lot_id.id)
+
+            line.prod_lot_ids = [(6, 0, list(lots))]
```

Comment:
```suggestion
            line.prod_lot_ids = line.mapped("sale_line_ids.move_ids.move_line_ids.lot_id")
```
This will have the same impact as below lines.